gnucash/import-export/qif-imp/qif-file.scm

```diff
@@ -525,18 +507,32 @@
       private-retval))
 
 
-  (gnc:backtrace-if-exception
+  (catch #t
     (lambda ()
-      (let ((retval #f))
+      (let ((retval #f)
+            (conv (port-conversion-strategy #f)))
+
+        (set-port-conversion-strategy! #f 'error)
         ;; Safely read the file.
-        (set! retval (gnc:backtrace-if-exception private-read))
+        (set! retval (private-read))
+        (set-port-conversion-strategy! #f conv)
 
         ;; Fill the progress dialog.
-        (if (and progress-dialog
-                 (list? retval))
+        (when (and progress-dialog (list? retval))
           (gnc-progress-dialog-set-value progress-dialog 1))
 
-        retval))))
+        retval))
+
+    (lambda (key . rest)
+      (let ((msg (string-append
+                  (case key
+                    ((decoding-error)
+                     (format #f (_ "Invalid charset ~a specified.") encoding))
```

Comment:
"Invalid charset" implies that `encoding` contains a value that's not a legal charset, and that's not likely since it's selected with a chooser that should contain only valid charset names. It also doesn't really explain to the user what is the problem. Something like "The file contains values that aren't part of charset ~a. Please choose a different one." would be better.